.github/workflows/_charm-tests-integration.yaml

```diff
@@ -32,6 +32,12 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 1
+      - name: Setup Charmcraft's pip cache
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/snap/charmcraft/common/cache/charmcraft/
+          key: charmcraft-cache-${{ github.job }}-${{ strategy.job-index }}-${{ github.run_id }}-${{ github.run_attempt }}
```

Comment:
Why do we need the key to be unique at all? If a the dep is there, it's there.

Reply:
Doing something like
```yaml
steps:
  - name: Restore cache
    uses: actions/cache/restore@v4
    with:
      path: ...
      key: charmcraft-cache
  - name: Integration tests
      ...
      ...
      ...
  - name: Save cache
    uses: actions/cache/save@v4
    with:
      path: ...
      key: charmcraft-cache
```

I think something like this would also work; what do you think?

Reply:
Apparently that might now work, because `save` might need a unique id (or it won't update the cache)

Reply:
I'm 99% sure that you cannot reuse a key in `cache/save` because github's cache items are immutable.  You can only delete, not update them.  I can't find the entry in github's docs, but [here](https://github.com/actions/toolkit/issues/505) is an open feature request asking for the ability to update because it isn't currently available 

The intent behind saving and restoring the cache is that you save as something very unique, then fuzzy match on restore